packages/reporter/src/hooks/hook-model.js

```diff
@@ -32,7 +34,16 @@ export default class Hook {
       return aliases.indexOf(alias) === i && aliases.lastIndexOf(alias) !== i
     })
 
-    return consecutiveDuplicateAliases.concat(nonConsecutiveDuplicateAliases)
+    const aliasesWithDuplicates = consecutiveDuplicateAliases.concat(nonConsecutiveDuplicateAliases)
+
+    // do a deep compare here to see if we can use the cached aliases, which will allow mobx's
+    // @computed identity comparison to pass, preventing unnecessary re-renders
+    // https://github.com/cypress-io/cypress/issues/4411
+    if (!_.isEqual(aliasesWithDuplicates, this._aliasesWithDuplicatesCache)) {
```

Comment:
I wanted to use `comparer.structural` here (https://mobx.js.org/refguide/computed-decorator.html#builtin-comparers) but it doesn't seem to accomplish the same thing.